meerk40t/tools/livinghinges.py

```diff
@@ -0,0 +1,1129 @@
+                        if new_cx >= xmin and new_cx<=xmax:
+                            new_cy = min(max(new_cy, ymin), ymax)
+                            new_ey = min(max(new_ey, ymin), ymax)
+                            if new_cx != current_x or new_cy != current_y:
+                                # Needs a move
+                                add_move(newpath, Point(new_cx, new_cy))
+                            newpath.line(Point(new_ex, new_ey))
+                            partial += 1
+                        else:
+                            ignored += 1
+                    else:
+                        # regular line, so lets establish x0 x1
+                        # could still be an outward pointing line....
+                        new_cx = min(max(new_cx, xmin), xmax)
+                        new_ex = min(max(new_ex, xmin), xmax)
+                        # corresponding y values...
+                        new_cy = current_y + (new_cx - current_x) / (p[0] - current_x) * (p[1] - current_y)
+                        new_ey = current_y + (new_ex - current_x) / (p[0] - current_x) * (p[1] - current_y)
+                        # Y can still cross...
+                        new_cx_clipped = new_cx
+                        new_ex_clipped = new_ex
+                        new_cy_clipped = min(max(new_cy, ymin), ymax)
+                        new_ey_clipped = min(max(new_ey, ymin), ymax)
+                        # Adjust x - value
+                        if dy != 0:
+                            new_cx_clipped = new_cx + dx / dy * (new_cy_clipped - new_cy)
+                            new_ex_clipped = new_ex + dx / dy * (new_ey_clipped - new_ey)
+
+                        new_cx = new_cx_clipped
+                        new_cy = new_cy_clipped
+                        new_ex = new_ex_clipped
+                        new_ey = new_ey_clipped
+                        if min(new_cy, new_ey) == ymax and dy != 0:
+                            # Outward...
+                            ignored
+                        elif max(new_cy, new_ey) == ymin and dy != 0:
+                            # Outward...
+                            ignored += 1
+                        else:
+                            if new_cx != current_x or new_cy != current_y:
+                                # Needs a move
+                                add_move(newpath, Point(new_cx, new_cy))
+                            newpath.line(Point(new_ex, new_ey))
+                            partial += 1
+                current_x= p[0]
+                current_y = p[1]
+            if current_x != part_of_path.end[0] or current_y != part_of_path.end[1]:
+                add_move(newpath, part_of_path.end)
+            # print (f"From iterated line: added={added}, partial={partial}, ignored={ignored}")
+
+        def add_move(addpath, destination):
+            # Was the last segment as well a move? Then just update the coords...
+            if len(addpath._segments)>0:
```

Comment:
You can access this through the path itself. `len(addpath)` is equal to this also `addpath[-1]` is equal. to the segments bits.